### PR TITLE
[infra] reproduce command

### PR DESCRIPTION
**Pull request by mikea**
This is an experimental version of reproduce command. I was thinking that CF could show display it on testcase page. Do you like it? Maybe we should use something else rather then base64? Maybe gzip it first?

(real) command line example:

```bash
docker run -ti ossfuzz/libxml2 reproduce libxml2_xml_read_memory_fuzzer PHhtbC8+Cg==
```

### CODE REVIEW

**Comment by oliverchang** [comment]
This would be a bit ugly with larger testcases right (assuming gzip doesn't make it much better)? 

e.g. a 60K testcase would result in 80kish characters in the commandline.

How about we just add this functionality to the helper script and make the developer download the testcase to somewhere first?

**Comment by mikea** [comment]
that would require a checkout of `oss-fuzz`. While it is ultimately ok, I'll think about the way to avoid this step.

**Comment by mikea** [comment]
@oliverchang I have replaced base64 argument with reading input from `stdin`. Full reproduce command now looks like:

```bash
cat fuzz-2-libxml2_xml_read_memory_fuzzer | docker run --rm ossfuzz/libxml2 reproduce libxml2_xml_read_memory_fuzzer
```

I think it now looks pretty neat. How about you?

**Comment by mikea** [comment]
I also talked to @kcc and he prefers something like:

```bash
docker run -v <local_file>:/testcase -t ossfuzz/libxml2 reproduce libxml2_xml_read_memory_fuzzer
```

**Comment by mikea** [comment]
Updated to use `/testcase`. PTAL.

File: infra/base-images/base-libfuzzer/reproduce [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# limitations under the License.
+#
+################################################################################
+
+FUZZER=$1
+shift
+TESTCASE="/testcase"
+
+if [ ! -f $TESTCASE ]; then
+  eho "Error: $TESTCASE not found, use: docker run -v <path>:$TESTCASE ..."
```
Comment (oliverchang) [inline infra/base-images/base-libfuzzer/reproduce]: s/eho/echo/ ?

File: infra/base-images/base-libfuzzer/reproduce [hunk 2]
```diff
@@ -15,16 +15,23 @@
...
 cd /out
-FUZZER=$1
-shift
-TEST_INPUT=$(cat -)
 
 TEST_CORPUS=$(tempfile)
 rm -rf $TEST_CORPUS
 mkdir $TEST_CORPUS
-echo $TEST_INPUT | $TEST_CORPUS/input
+cp $TESTCASE $TEST_CORPUS/
```
Comment (oliverchang) [inline infra/base-images/base-libfuzzer/reproduce]: why do we need a directory? can we just pass the testcase path directly?

File: infra/base-images/base-libfuzzer/reproduce (hunk 1 above)
Comment (mikea) [inline infra/base-images/base-libfuzzer/reproduce]: Done

File: infra/base-images/base-libfuzzer/reproduce (hunk 2 above)
Comment (mikea) [inline infra/base-images/base-libfuzzer/reproduce]: No, we can't. You can only pass a corpus directory. Once we use https://reviews.llvm.org/rL285135 for reproducing, we'll get rid of directory.

File: infra/base-images/base-libfuzzer/reproduce [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+compile
+export PATH=/out:$PATH
+cd /out
+
+TEST_CORPUS=$(tempfile)
+rm -rf $TEST_CORPUS
+mkdir $TEST_CORPUS
+cp $TESTCASE $TEST_CORPUS/
+$FUZZER $@ -runs=0 $TEST_CORPUS
```
Comment (oliverchang) [inline infra/base-images/base-libfuzzer/reproduce]: (re-adding comment, looks like my other one was hidden by default)

why do we need a directory? can we just pass the testcase path directly?

File: infra/base-images/base-libfuzzer/reproduce (hunk 2 above)
Comment (mikea) [inline infra/base-images/base-libfuzzer/reproduce]: Of course it works, my bad. I saw some strange error message before. PTAL.

Comment (oliverchang) [inline infra/base-images/base-libfuzzer/reproduce]: super nit: is the -runs=0 necessary?

LGTM otherwise, thanks!

**Pull request merged**